contents/tutorials/framer-ab-tests.md

```diff
@@ -0,0 +1,132 @@
+date: 2023-11-09
+author: ["lior-neu-ner"]
+showTitle: true
+sidebar: Docs
+tags: ['experimentation', 'feature flags']
+---
+
+[Framer](https://www.framer.com/) is a great tool for building marketing websites. However, sometimes you may be unsure if a change you're making will actually improve your conversion rate. This is where [A/B testing](/ab-testing) is helpful. It enables you to test and compare your changes.
+
+This tutorial shows you how to set up A/B tests with Framer and PostHog to get the most out of your website.
+
+## Adding PostHog to your Framer site
+
+First, [sign up to PostHog](https://app.posthog.com/signup). Then, go to your [project settings](https://app.posthog.com/settings/project) and copy your web snippet:
+
+![PostHog web snippet](../images/tutorials/framer-ab-tests/web-snippet.png)
+
+With the snippet copied, go to your Framer project settings by clicking the gear in the top right. If you haven’t already, sign up for the "Mini" site plan. This enables you to add custom code.
+
+Go to the "General" tab in site settings and scroll down to the "Custom Code" section. Under "End of `<head>` tag", paste your PostHog snippet there. Make sure to press "Save" next to custom code.
+
+![Script](../images/tutorials/framer-analytics/script.mp4)
+
+## Adding a custom event
+
+We're going to create an A/B test comparing how the text of a button affects it's click-through rate.
+
+To do this, first we'll create a button using [custom code components](https://www.framer.com/developers/#code-components). Then, we [capture an event](/docs/product-analytics/capture-events) whenever the button is clicked. We'll use this event as our [goal metric](/product-engineers/ab-testing-guide-for-engineers#1-a-clear-measurable-goal) for our A/B test.
+
+Go to the "Assets" tab in the top left of your Framer project. Then, click the plus icon next to the "Code" tab. This will show a pop up to create a new code file. Name the file `CaptureButton`, set it as a "New component" and press "Create."
+
+In the new code file, delete the existing code and replace it with the following:
+
+```js
+export default function CaptureButton() {
+    const handleClick = () => {
+        window.posthog.capture("clicked_homepage_button", {
+            $set_once: { clicked_homepage_button: true },
+        })
+    }
+
+    return (
+        <button id="capture-button" onClick={handleClick}>
+            Click me
+        </button>
+    )
+}
+```
+
+Press `Cmd/Ctrl + s` to save your changes. Then press the "Home" button to go back to the home page. Add your new 
+`CaptureButton` to your page by dragging it from the Code tab.
+
+![Create a code snippet in Framer](../images/tutorials/framer-ab-tests/create-code-component.mp4)
+
+
+Publish your site and then click your new button to [see the event in PostHog](https://app.posthog.com/events).
+
+![View captured events in PostHog](../images/tutorials/framer-ab-tests/event-captured.png)
```

Comment:
probably just I will notice this, but it is a bit weird to have the autocaptured events not match the button text at this point. looks like it is autocaptured later (once the A/B test is running, and the text has changed)